packages/grid/_modules_/grid/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -0,0 +1,252 @@
+        return {
+          ...state,
+          containerSizes: {
+            ...state.containerSizes!,
+            renderingZone: {
+              ...state.containerSizes!.renderingZone,
+              height: visibleSortedRows.size * props.rowHeight!,
+            },
+            renderingZonePageSize: visibleSortedRows.size,
+          },
+          viewportSizes: {
+            ...state.viewportSizes,
+            height: visibleSortedRows.size * props.rowHeight!,
+          },
+        };
+      });
+      forceUpdate();
+    },
+    [columns, visibleSortedRows, props.rowHeight, apiRef, setGridState, forceUpdate],
+  );
+
+  const buildPrintWindow = React.useCallback((title?: string): HTMLIFrameElement => {
+    const iframeEl = document.createElement('iframe');
+
+    iframeEl.style.position = 'absolute';
+    iframeEl.style.width = '0px';
+    iframeEl.style.height = '0px';
+    iframeEl.title = title || document.title;
+
+    return iframeEl;
+  }, []);
+
+  const handlePrintWindowLoad: PrintWindowOnLoad = React.useCallback(
+    (printWindow, options): void => {
+      const normalizeOptions = {
+        copyStyles: true,
+        hideToolbar: false,
+        hideFooter: false,
+        ...options,
+      };
+
+      // Some agents, such as IE11 and Enzyme (as of 2 Jun 2020) continuously call the
+      // `onload` callback. This ensures that it is only called once.
+      printWindow.onload = null;
+
+      const printDoc = printWindow.contentDocument || printWindow.contentWindow?.document;
+
+      if (!printDoc) {
+        return;
+      }
+
+      const gridRootElement = apiRef!.current.rootElementRef!.current;
+      const gridClone = gridRootElement!.cloneNode(true) as HTMLElement;
+
+      if (normalizeOptions.hideToolbar) {
+        gridClone.querySelector('.MuiDataGrid-toolbarContainer')?.remove();
+      }
+
+      if (normalizeOptions.hideFooter) {
+        gridClone.querySelector('.MuiDataGrid-footerContainer')?.remove();
+      }
+
+      // Expand container height to accommodate all rows
+      gridClone.style.height = `${visibleSortedRows.size * props.rowHeight!}px`;
+      printDoc.body.appendChild(gridClone);
+
+      const defaultPageStyle =
+        typeof normalizeOptions.pageStyle === 'function'
+          ? normalizeOptions.pageStyle()
+          : normalizeOptions.pageStyle;
+      if (typeof defaultPageStyle !== 'string') {
+        const styleElement = printDoc.createElement('style');
+        styleElement.appendChild(printDoc.createTextNode(defaultPageStyle));
+        printDoc.head.appendChild(styleElement);
+      }
+
+      if (normalizeOptions.bodyClass) {
+        printDoc.body.classList.add(...normalizeOptions.bodyClass.split(' '));
+      }
+
+      if (normalizeOptions.copyStyles) {
+        const headStyleElements = doc.current!.querySelectorAll("style, link[rel='stylesheet']");
+
+        // eslint-disable-next-line no-plusplus
+        for (let i = 0; i < headStyleElements.length; i++) {
```

Comment:
```suggestion
        for (let i = 0; i < headStyleElements.length; i+=1) {
```

Otherwise we should remove the linting rule if we start disabling it.
I'm not a fan of this `no plus plus` rule but as you want :+1: